tools/debug/debugsymbolviewer.py

```diff
@@ -0,0 +1,178 @@
+		print('~~~~~~~~~~~~~~~~~~~~~~~~~~  KERNEL TEXT RANGE ~~~~~~~~~~~~~~~~~~~~~~~')
+		print('')
+
+
+# Function to search the search_addr present in the System map file and print its corresponding symbol
+def print_symbol(search_addr):
+	addr = 0x0
+	next_addr = 0x0
+	l = 0
+	h = len(symbol_lookup_table)
+	mid = (l + h) / 2
+
+	while (l <= h):
+		addr = symbol_lookup_table[mid - 1][0]
+		next_addr = symbol_lookup_table[mid][0]
+
+		if (search_addr >= addr and search_addr <= next_addr):
+			print("Address:  {1}\tSymbol:  {0}".format(symbol_lookup_table[mid - 1][1], hex(addr)))
+			break
+
+		if (search_addr < addr):
+			h = mid - 1
+		else:
+			l = mid + 1
+
+		mid = (l + h) / 2
+
+	if (l > h):
+		print('Symbol not found in system map file')
+
+
+# Function to check if address lies in the kernel text address range
+def is_kernel_text_address(address):
+
+	# Check the address
+	if (address >= ram_text_start_addr and address < ram_text_end_addr) or (address >= flash_text_start_addr and address < flash_text_end_addr):
+		if debug:
+			print("{0} is kernel text address!".format(address))
+		return True
+	else:
+		return False
+
+
+# Function to Parse the i/p log file (which contains stackdump during assert)
+def parse_file(log_file):
+	k = 1
+	i = 0
+	got = 0
+	offset = 0
+	# Parse the contents based on tokens in log file.
+	with open(log_file) as searchfile:
+		for line in searchfile:
+			# Get the stack size
+			if 'up_dumpstate:' in line:
+				word = re.split(r'[:]',line)
+				if word[1] == ' User stack':
+					continue
+				if word[1] == '   size':
+					# word[2] contains the stack size
+					stacksize = int(word[2],16)
+				continue
+			if 'up_registerdump:' in line:
+				word = line.split(':')
+				if word[1] == ' R8':
+					t = word[2].split( )
+					# Last subword of word[2] contains the PC value
+					for sub_word in t:
+						pc_value = (int(sub_word,16))
+					print('\nPC value 0x{0:x} corresponds to:'.format(pc_value))
+					print_symbol(pc_value)
+				continue
+			# Read the stack contents of aborted stack and Populate stack_table
+			if 'up_stackdump:' in line:
+				word = line.split(':')
+				t = word[2].split( )
+				if k == 1:
+					# word[1] contains the stack start address
+					stack_start_addr = int(word[1], 16)
+					k = 0
+				for sub_word in t:
+					stack_table.append((stack_start_addr,int(sub_word,16)))
+					stack_start_addr = stack_start_addr + 4
+
+					# Check if the stack address lies in kernel text address range
+					if (is_kernel_text_address(hex(stack_table[i][1]))):
+						#If yes, print it's corresponding symbol
+						print_symbol((stack_table[i][1]))
+
+					i = i + 1
+
+#Excution starts here
+if (__name__ == '__main__'):
+    # init
+
+    setup_symbol_table(systemmap_path)
+
+    print('~~~~~~~~~~~~~~~~~~~~~~~~  DEBUG DISPLAY SYMBOLS ~~~~~~~~~~~~~~~~~~~~~')
+    parse_file(log_file)
+    print('~~~~~~~~~~~~~~~~~~~~~~~~  DEBUG DISPLAY SYMBOLS ~~~~~~~~~~~~~~~~~~~~~')
```

Comment:
I think this line is not needed. Or we don't need to print the message in this line. Just `~~~` is enough.